Utilities/Mergers/test/customTopologyCommon.h

```diff
@@ -0,0 +1,184 @@
+      "data-checker",
+      Inputs{
+        {"custom", origin, description, 0, Lifetime::Sporadic},
+        {"custom_mw", origin, description_moving_window, 0, Lifetime::Sporadic},
+      },
+      Outputs{},
+      AlgorithmSpec{
+        AlgorithmSpec::InitCallback{[expectedResult = mExpectedResult](InitContext&) {
+          return AlgorithmSpec::ProcessCallback{
+            [expectedResult, numberOfCalls = 0, numberOfObjects = 0, numberOfMovingWindows = 0, lastObjectValue = 0](ProcessingContext& processingContext) mutable {
+              numberOfCalls++;
+
+              if (processingContext.inputs().isValid("custom")) {
+                auto obj = processingContext.inputs().get<mergers::CustomMergeableObject*>("custom");
+                numberOfObjects++;
+                // we are keeping only the last value of secret, as there can be inconsitencies caused
+                // by the lack of synchronisation between merger layers
+                lastObjectValue = obj->getSecret();
+              }
+
+              if (processingContext.inputs().isValid("custom_mw")) {
+                auto mw = processingContext.inputs().get<mergers::CustomMergeableObject*>("custom_mw");
+                numberOfMovingWindows++;
+                // it is not possible to check for correct value in moving window as the value can be lost due
+                // to the sync between layers of mergers and movement of the window
+              }
+
+              if (numberOfCalls == 5) {
+                processingContext.services().get<ControlService>().readyToQuit(QuitRequest::All);
+
+                if (numberOfObjects != 5 || numberOfMovingWindows == 0 || numberOfMovingWindows >= 3) {
+                  LOG(fatal) << "expected 5 objects and got: " << numberOfObjects << ", expected 1 or 2 moving windows and got: " << numberOfMovingWindows;
+                  if (lastObjectValue != expectedResult) {
+                    LOG(fatal) << "got wrong secret from object: " << lastObjectValue;
+                  }
+                }
+              }
+            }};
+        }}}});
+  }
+
+  void generateCheckerFullHistory(WorkflowSpec& specs)
+  {
+    specs.push_back(DataProcessorSpec{
+      "data-checker",
+      Inputs{
+        {"custom", origin, description, 0, Lifetime::Sporadic},
+      },
+      Outputs{},
+      AlgorithmSpec{
+        AlgorithmSpec::InitCallback{[expectedResult = mExpectedResult](InitContext&) {
+          return AlgorithmSpec::ProcessCallback{
+            [expectedResult, retryNumber = 0, numberOfRetries = 5](ProcessingContext& processingContext) mutable {
+              const auto obj = processingContext.inputs().get<mergers::CustomMergeableObject*>("custom");
+              if (obj->getSecret() == expectedResult) {
+                processingContext.services().get<ControlService>().readyToQuit(QuitRequest::All);
+                return;
+              } else {
+                if (retryNumber++ == numberOfRetries) {
```

Comment:
```suggestion
                if (retryNumber++ == maxNumberOfRetries) {
```
Please rename for clarity